File: bk2ch06p272forcedRotation/SecondViewController.m

#import "SecondViewController.h"

@interface SecondViewController ()

@end

@implementation SecondViewController

- (IBAction)doDismiss:(id)sender {
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void) viewDidAppear:(BOOL)animated {
    [super viewDidAppear: animated];
    // prove that you've got data
    NSLog(@"%@", self.data);
}

- (void) viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear: animated];
    if ([self isBeingDismissed])
        [self.delegate acceptData: @"Even more important data!"];
}

-(NSUInteger)supportedInterfaceOrientations {
    NSLog(@"%@", @"presented supported");
    return UIInterfaceOrientationMaskLandscape;
}

#define which 0
#if which == 1

-(UIInterfaceOrientation)preferredInterfaceOrientationForPresentation {
    return UIInterfaceOrientationLandscapeLeft;
}

#endif

-(void)viewWillLayoutSubviews {
    NSLog(@"%@", @"presented will layout");
}


@end

File: iOS9bookExamples/bk2ch06p272forcedRotation/ch19p601presentedViewController/SecondViewController.swift


import UIKit


protocol SecondViewControllerDelegate : class {
    func acceptData(data:AnyObject!)
}

class SecondViewController : UIViewController {
    
    var data : AnyObject?
    
    weak var delegate : SecondViewControllerDelegate?
    
    @IBAction func doDismiss(sender:AnyObject?) {
        self.presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        // prove you've got data
        print(self.data)
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        if self.isBeingDismissed() {
            self.delegate?.acceptData("Even more important data!")
        }
    }
    
    /* 
    ability of presented view controller to force app rotation
    missing in initial seeds of iOS 8, restored in seed 4
*/

    override func shouldAutorotate() -> Bool {
        return true
    }
    
    override func supportedInterfaceOrientations() -> UIInterfaceOrientationMask {
        print("second supported")
        return .Landscape
    }
    
    override func preferredInterfaceOrientationForPresentation() -> UIInterfaceOrientation {
        return .LandscapeLeft
    }
    
    override func viewWillLayoutSubviews() {
        print("presented will layout")
    }
    
    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
        print("presented size")
        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
    }

    
}
